java-support/src/main/java/io/cloudstate/javasupport/EntitySupportFactory.java

```diff
@@ -0,0 +1,18 @@
+package io.cloudstate.javasupport;
+
+/** Auxiliar interface for support DI environments */
+public interface EntitySupportFactory {
+  /**
+   * The reference to the Entity.
+   *
+   * @return The reference to the Entity.
+   */
+  Object create(Context context, String enittyId);
```

Comment:
Can't the return type be more specific than `Object`?

Reply:
Hummm I don't know. Unless user entities implement some type of interface, I think it would not be possible to determine the correct type.